### PR TITLE
Enable RSpec/DescribeClass

### PR DESCRIPTION
**Pull request by dduugg**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----
Follow-up to https://github.com/Homebrew/brew/pull/16815/files#r1518269399

Disabling this cop made sense, when we would have had ~100 command spec files using `Homebrew`, but now they're each individually namespaced.

There are still a few stray files that don't make sense to spec under a class. For the most part, I used the [escape hatch](https://docs.rubocop.org/rubocop-rspec/cops_rspec.html#configurable-attributes-6) of labeling them `type: system`. That's arguably too esoteric though, and perhaps I should just `rubocop:disable` them instead (but i consider the latter to be a bit of an eyesore). Feedback welcome.

### CODE REVIEW

File: Library/Homebrew/test/cask/conflicts_with_spec.rb [hunk 1]
```diff
@@ -1,25 +0,0 @@
-# frozen_string_literal: true
-
-RSpec.describe "conflicts_with", :cask do
```
Comment (dduugg) [inline Library/Homebrew/test/cask/conflicts_with_spec.rb:3]: It seemed weird to have this in its own file – I moved this to `dsl_spec`, where there are similar tests.

File: Library/Homebrew/test/cask/depends_on_spec.rb [hunk 2]
```diff
@@ -2,7 +2,9 @@
 
 # TODO: this test should be named after the corresponding class, once
 #       that class is abstracted from installer.rb
+# rubocop:disable RSpec/DescribeClass
```
Comment (dduugg) [inline Library/Homebrew/test/cask/depends_on_spec.rb:5]: This would resolve with the TODO above it.

File: Library/Homebrew/test/patching_spec.rb [hunk 3]
```diff
@@ -229,15 +229,3 @@ def formula(name = "formula_name", path: Formulary.core_path(name), spec: :stabl
     end.to raise_error(BuildError)
   end
 end
-
-__END__
```
Comment (dduugg) [inline Library/Homebrew/test/patching_spec.rb:233]: Unrelated cleanup

**Review by MikeMcQuaid — COMMENTED** [review]
Looks good, thanks again @dduugg!
